cpp/include/cudf/join.hpp

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+namespace cudf {
+// joins
+
+/** 
+ * @brief  Performs an inner join on the specified columns of two
+ * tables (left, right)
+ *
+ * @example TableA a:{0,1,2}
+ *          TableB b:{1,2,3}, a:{1,2,5} And column 'a' from TableB
+ *          joins column 'a' from TableA.
```

Comment:
```suggestion
 *          columns_in_common: { {0, 0} }
```

Reply:
It would probably be good to also show what the result would be if `columns_in_common` were empty.